src/execution_plan/execution_plan.c

```diff
@@ -658,26 +658,18 @@ ExecutionPlan* NewExecutionPlan(RedisModuleCtx *ctx, AST **ast, ResultSet *resul
     return plan;
 }
 
-void _ExecutionPlanPrint(const OpBase *op, char **strPlan, int ident) {
-    char strOp[512] = {0};
-    sprintf(strOp, "%*s%s\n", ident, "", op->name);
-    
-    if(*strPlan == NULL) {
-        *strPlan = calloc(strlen(strOp) + 1, sizeof(char));
-    } else {
-        *strPlan = realloc(*strPlan, sizeof(char) * (strlen(*strPlan) + strlen(strOp) + 2));
-    }
-    strcat(*strPlan, strOp);
+void _ExecutionPlan_Print(const OpBase *op, RedisModuleCtx *ctx) {
+    if(!op) return;
 
+    RedisModule_ReplyWithArray(ctx, op->childCount + 1);
```

Comment:
I think this causes less readable output than an indentation-structured approach. I think the @stockholmux format (which just replaces our newline bytes with separate array members) is very clear:
```
"MATCH (r:Rider)-[:rides]->(t:Team) WHERE t.name = 'Yamaha' RETURN r,t"
1) Results
2)    Project
3)        Conditional Traverse
4)            Filter
5)                Node By Label Scan
```
The nested-array approach here produces the output:
```
1) Results
2) 1) Project
   2) 1) Conditional Traverse
      2) 1) Filter
         2) 1) Node By Label Scan
```
Which I think is less clear, and somewhat misleading.

Reply:
How about this:
`GRAPH.EXPLAIN "MATCH (N),(M:L) WHERE M.v = 1 RETURN N"
1) "Results"
2) "  Project"
3) "    Cartesian Product"
4) "      All Node Scan"
5) "      Filter"
6) "        Node By Label Scan"`

Reply:
Much cleaner, yeah!